src/main/java/net/openhft/chronicle/queue/impl/single/SingleChronicleQueue.java

```diff
@@ -1216,6 +1216,10 @@ public int nextCycle(int currentCycle, @NotNull TailerDirection direction) {
                 fileFound |= currentCycleFile.exists();
 
                 if (!fileFound) {
+                    if (currentCycle < directoryListing.getMinCreatedCycle()) {
+                        Jvm.warn().on(SingleChronicleQueue.class, "Current cycle has been deleted, resuming from first available cycle");
+                        return toCycle(tree.firstEntry());
+                    }
```

Comment:
In the specific case that we're on a roll cycle earlier than the minimum one, and the cycle we're on is no longer there, can we skip forward to the earliest existing cycle? We probably also need to handle the case where there are all of a sudden no roll cycles. I'll beef up the testing around that.